utils/ssh.py

```diff
@@ -219,33 +255,34 @@ def run_command(self, command, timeout=RUNCMD_TIMEOUT, reraise=False, ensure_hos
             while True:
                 if session.recv_ready:
                     for line in stdout:
-                        output += line
+                        output.append(line)
                         if self._streaming:
                             sys.stdout.write(line)
 
                 if session.recv_stderr_ready:
                     for line in stderr:
-                        output += line
+                        output.append(line)
                         if self._streaming:
                             sys.stderr.write(line)
 
                 if session.exit_status_ready():
                     break
             exit_status = session.recv_exit_status()
-            return SSHResult(exit_status, output)
+            return SSHResult(exit_status, ''.join(output))
         except paramiko.SSHException as exc:
             if reraise:
                 raise
             else:
                 logger.exception(exc)
         except socket.timeout as e:
-            logger.error("Command `%s` timed out.", command)
+            logger.error("Command `{command}` timed out.".format(command=command))
             logger.exception(e)
-            logger.error("Output of the command before it failed was:\n%s", output)
+            logger.error("Output of the command before it failed was:\n{output}".format(
+                output=''.join(output)))
             raise
 
         # Returning two things so tuple unpacking the return works even if the ssh client fails
-        return SSHResult(1, None)
+        return SSHResult(1, '')
```

Comment:
I'm a little concerned about this change...it's so tiny it could just break things in new and interesting ways, what's the rationale?